services/core/shared/core/compat/sdk_v5/transactions.js

```diff
@@ -208,7 +208,34 @@ const getTransactions = async params => {
 	return transactions;
 };
 
+const getTransactionsByBlockId = async blockId => {
+	const [block] = (await coreApi.getBlockByID(blockId)).data;
+	const transactions = await BluebirdPromise.map(
+		block.payload,
+		async transaction => {
+			transaction.unixTimestamp = block.header.timestamp;
+			transaction.height = block.header.height;
+			transaction.blockId = block.header.id;
+			const account = await getIndexedAccountInfo({ publicKey: transaction.senderPublicKey });
```

Comment:
Is there any reason to get account by public key instead of address?

Reply:
There is no specific reason here. The only reason is we are retrieving blocks directly from `lisk-core` and sender address is not available in `block.payload`.  We can resolve `address` from `public key` and then get `account` by address but anyway, we are resolving missing information in transaction for specific account(username & address).
```
 const account = await getIndexedAccountInfo({ publicKey: transaction.senderPublicKey });
 transaction.senderId = account && account.address ? account.address : undefined;
 transaction.username = account && account.username ? account.username : undefined;
```